src/qmxgraph/page/api.js

```diff
@@ -1125,6 +1125,56 @@ graphs.Api.prototype.removeCells = function removeCells(cellIds, ignoreMissingCe
     graph.removeCells(cells);
 };
 
+/**
+ * Clone cells and add them to the graph.
+ *
+ * @param {number[]} cellIds Ids of cells to be cloned.
+ * @param {boolean} ignoreMissingCells Ids of non existent cells are ignored instead of
+ * raising an error.
+ * @param {boolean} allowInvalidEdges Specifies if invalid edges should be cloned.
```

Comment:
I confess I'm not quite sure. I'm just surfacing the underlying `mxGraph` API. But since I'm already changing the behaviour by implicitly adding the cells to the graph on `cloneCells` (which `mxGraph` doesn't do), then maybe I could just ommit that parameter since it's not useful to us.

Reply:
Update: reading the `mxGraph`'s source code just now, it seems that "invalid edges" is all about the edges' terminal nodes. It checks for dangling edges, loops, edges missing both terminal nodes, etc.